cmd/gbounty/bootstrap/update.go

```diff
@@ -22,9 +23,41 @@ import (
 	"github.com/bountysecurity/gbounty/kit/slices"
 )
 
+func getLastCheckTime() (time.Time, error) {
+	filePath, err := lastCheckFilePath()
+	if err != nil {
+		return time.Time{}, err
+	}
+
+	data, err := os.ReadFile(filePath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return time.Time{}, nil
+		}
+		return time.Time{}, err
+	}
+
+	return time.Parse(time.RFC3339, string(data))
+}
+
+func checkForUpdatesRequired() bool {
+	const updateInterval = 24 * time.Hour
+	lastCheckTime, err := getLastCheckTime()
+	currentTime := time.Now()
+
+	if err != nil || lastCheckTime.IsZero() || currentTime.Sub(lastCheckTime) > updateInterval {
+		return true
+	}
+	return false
+}
+
 func CheckForUpdates() {
-	// Disable checks on CI jobs.
-	if _, isCI := os.LookupEnv("CI"); isCI {
+	// Disable checks on CI jobs and when updates not required.
+	if _, isCI := os.LookupEnv("CI"); isCI || !checkForUpdatesRequired() {
```

Comment:
⚠️ This breaks the updates.

You can easily reproduce the user journey with:
1. You run `gbounty`, you see the available updates.
2. You run `gbounty --update`, it does nothing (which is wrong 🔴)

Reply:
To correct that, you'll need to take into account if any updated flag is supplied (see the code a bit below in the function).

Reply:
Btw, I'd also add a `--check-updates` flag, that invalidates this behavior of checking updates only once per day, in case the user wants to explicitly check it.

Reply:
Hi @joanlopez, for case of `--update` flag is passed should we update last check file ? if no then its behaviour it will be equivalent to `--check-updates` flag ?

Reply:
I don't think it matters a lot, but yeah, I guess that if the `--update` flag is passed, we likely want to update the last check file, so after the update, it isn't checked any more until next day (unlikely a new release is published within the same day). Does that make sense? 🤔

Reply:
yes right. thanks